services/src/main/scala/cromwell/pipeline/service/HttpClient.scala

```diff
@@ -1,17 +1,30 @@
 package cromwell.pipeline.service
 
+import play.api.libs.json.{ Reads, Writes }
+
 import scala.concurrent.{ ExecutionContext, Future }
 
 trait HttpClient {
-  def get(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
-    implicit ec: ExecutionContext
-  ): Future[Response]
-  def post(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: String)(
-    implicit ec: ExecutionContext
-  ): Future[Response]
-  def put(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: String)(
-    implicit ec: ExecutionContext
-  ): Future[Response]
+  def get[B](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
+    implicit ec: ExecutionContext,
+    f: Reads[B]
+  ): Future[Response[B]]
+  def post[B, P](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: P)(
+    implicit ec: ExecutionContext,
+    bf: Reads[B],
+    pf: Writes[P]
+  ): Future[Response[B]]
+  def put[B, P](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: P)(
+    implicit ec: ExecutionContext,
+    bf: Reads[B],
+    pf: Writes[P]
+  ): Future[Response[B]]
 }
 
-case class Response(status: Int, body: String, headers: Map[String, Seq[String]])
+case class Response[B](status: Int, body: ResponseBody[B], headers: Map[String, Seq[String]])
+
+sealed trait ResponseBody[B]
+
+final case class PositiveResponseBody[B](bodyObject: B) extends ResponseBody[B]
```

Comment:
Let's stick to scala naming conventions and use something like `success` and `failure` instead of `positive` and `negative`.

Reply:
done